src/partitioning/subdomain_partitioner.C

```diff
@@ -56,14 +57,12 @@ void SubdomainPartitioner::_do_partition (MeshBase & mesh,
 
   // For each chunk, construct an iterator range for the set of
   // subdomains in question, and pass it to the internal Partitioner.
-  for (std::size_t c=0; c<chunks.size(); ++c)
-    {
-      MeshBase::element_iterator
-        it = mesh.active_subdomain_set_elements_begin(chunks[c]),
-        end = mesh.active_subdomain_set_elements_end(chunks[c]);
-
-      _internal_partitioner->partition_range(mesh, it, end, n);
-    }
+  for (auto c : index_range(chunks))
```

Comment:
Do we want an index_range here?  Looks like a range for over chunks would work.

Reply:
Yes, will change.